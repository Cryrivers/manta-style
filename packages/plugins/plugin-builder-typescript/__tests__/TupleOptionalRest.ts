import getTranspiledString from '../src/utils/transpile';

describe('Tuple, Optional and Rest', () => {
  test('Tuple', () => {
    expect(
      getTranspiledString('type Test = [number, number]'),
    ).toMatchSnapshot();
  });
  test('Optional', () => {
    expect(
      getTranspiledString('type Test = [number, number, number?]'),
    ).toMatchSnapshot();
  });
  test('Rest', () => {
    expect(
      getTranspiledString('type Test = [number, number, ...string[]]'),
    ).toMatchSnapshot();
  });
});
