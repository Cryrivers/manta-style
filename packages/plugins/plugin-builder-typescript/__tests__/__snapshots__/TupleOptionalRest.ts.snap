// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuple, Optional and Rest Optional 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].TupleType([runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].OptionalType(runtime_1[\\"default\\"].NumberKeyword)]);
    return type;
}, []);
"
`;

exports[`Tuple, Optional and Rest Rest 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].TupleType([runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].RestType(runtime_1[\\"default\\"].StringKeyword)]);
    return type;
}, []);
"
`;

exports[`Tuple, Optional and Rest Tuple 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var Test = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"Test\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].TupleType([runtime_1[\\"default\\"].NumberKeyword, runtime_1[\\"default\\"].NumberKeyword]);
    return type;
}, []);
"
`;
